modules/ssl/ssl_engine_pphrase.c

```diff
@@ -600,3 +610,330 @@ int ssl_pphrase_Handle_CB(char *buf, int bufsize, int verify, void *srv)
      */
     return (len);
 }
+
+#if defined(HAVE_OPENSSL_ENGINE_H) && defined(HAVE_ENGINE_INIT)
+int passphrase_ui_open(UI *ui)
+{
+    pphrase_cb_arg_t *ppcb_arg;
+    SSLSrvConfigRec *sc;
+    
+    ppcb_arg = (pphrase_cb_arg_t *) UI_get0_user_data(ui);
+
+    if (ppcb_arg == NULL) {
+        //TODO LOG Properly
+        return 0;
+    }
+
+    sc = mySrvConfig(ppcb_arg->s);
+
+    ppcb_arg->nPassPhraseDialog++;
+    ppcb_arg->nPassPhraseDialogCur++;
+    
+    /*
+     * Builtin or Pipe dialog
+     */
+    if (sc->server->pphrase_dialog_type == SSL_PPTYPE_BUILTIN
+            || sc->server->pphrase_dialog_type == SSL_PPTYPE_PIPE) {
+        if (sc->server->pphrase_dialog_type == SSL_PPTYPE_PIPE) {
+            if (!readtty) {
+                ap_log_error(APLOG_MARK, APLOG_INFO, 0, ppcb_arg->s,
+                        APLOGNO(01965)
+                        "Init: Creating pass phrase dialog pipe child "
+                        "'%s'", sc->server->pphrase_dialog_path);
+                if (ssl_pipe_child_create(ppcb_arg->p,
+                            sc->server->pphrase_dialog_path)
+                        != APR_SUCCESS) {
+                    ap_log_error(APLOG_MARK, APLOG_ERR, 0, ppcb_arg->s,
+                            APLOGNO(01966)
+                            "Init: Failed to create pass phrase pipe '%s'",
+                            sc->server->pphrase_dialog_path);
+                    //TODO fix this error
```

Comment:
what is the error you'd like to fix?

Reply:
This part of the code was copied from the existing UI callback implementation. I suppose the error was registered by PEMerr(), because originally the code was trying to open an encrypted PEM file. I have to find out the right way to register my error.